spec/controllers/courses/title_controller_spec.rb

```diff
@@ -0,0 +1,105 @@
+require "rails_helper"
+
+RSpec.describe Courses::TitleController do
+  let(:provider) { build(:provider) }
+  let(:course) { build(:course, provider: provider) }
+  let(:admin) do
+    {
+      user_id: 1,
+      uid: SecureRandom.uuid,
+      info: {
+        email: "dave@example.com",
+      },
+      admin: true,
+    }.with_indifferent_access
+  end
+
+  before do
+    allow(controller).to receive(:current_user).and_return(current_user)
+    stub_api_v2_resource(course.recruitment_cycle)
+    stub_api_v2_resource(course)
+  end
+
+  describe "#edit" do
+    context "when a non-admin" do
+      let(:current_user) do
+        {
+          user_id: 1,
+          uid: SecureRandom.uuid,
+          info: {
+            email: "dave@example.com",
+          },
+        }.with_indifferent_access
+      end
+
+      it "redirects to root" do
+        get :edit, params: {
+          provider_code: provider.provider_code,
+          recruitment_cycle_year: course.recruitment_cycle.year,
+          code: course.course_code,
+        }
+
+        expect(response).to redirect_to("/")
```

Comment:
when you are not an `admin` yet you try to do an `admin task` you should be `forbidden` not `redirect`

Reply:
Sure I can change that

Reply:
I contemplated redirecting them to forbidden but have decided to just render it instead

Reply:
rendered is prefect